### PR TITLE
Fix cypress auth test and test debugging improvements

### PR DESCRIPTION
**Pull request by ssangervasi**
Follow up to #1636

### Code Changes

- The actual fixes:
  - Intercept logout request
  - Wait for plus health on classify page
- Improvements to make cypress debugging easier:
  - Upload cypress recordings on test failure
  - Define environment variables for test
  - Disable rewrites in test to fix ECONNREFUSED log

### Steps to Confirm

* Check the Admin-UI-Cypress workflow

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation Updated:
  * [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  * [ ] documentation issue created (tag docs-team to complete issue separately)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
    - https://github.com/ethyca/fides/issues/1732
* [x] Update `CHANGELOG.md`

### Description Of Changes

The underlying issues were just requests that were slow or not stubbed, so would only pass locally. Debugging was CI just too hard to pinpoint quickly, so I included changes that should make it easier.

### CODE REVIEW

File: clients/admin-ui/package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "copy-export": "mkdir -p ../../src/fides/ui-build/static/admin/ && rsync -a --delete out/ ../../src/fides/ui-build/static/admin/",
     "cy:open": "cypress open",
     "cy:run": "cypress run",
-    "cy:start": "next build && NODE_ENV=test next start",
+    "cy:start": "NODE_ENV=test next build && NODE_ENV=test next start",
```
Comment (ssangervasi) [inline clients/admin-ui/package.json:13]: This, the `next.config` change, and the `.env.test` are all to cut down on the noise that makes debugging cypress failures harder.

File: .github/workflows/frontend_checks.yml [hunk 2]
```diff
@@ -73,6 +73,12 @@ jobs:
           wait-on: "http://localhost:3000"
           wait-on-timeout: 120
 
+      - uses: actions/upload-artifact@v3
```
Comment (ssangervasi) [inline .github/workflows/frontend_checks.yml:76]: @NevilleS - This uploads a <10mb zip on test failure. Based on the GH docs I don't think this pushes any data limits, but I figured I'd call it out.

Comment (allisonking) [reply .github/workflows/frontend_checks.yml:76]: where does this get uploaded to? also, will it upload all failing tests, or just the first one? (if all, then the zip could potentially be larger?)

File: clients/admin-ui/next.config.js [hunk 3]
```diff
@@ -21,6 +21,12 @@ const nextConfig = {
         return config;
     },
     async rewrites() {
+        // The tests run without a server. Rewrites cause Next to continually try to connect,
+        // which spams the logs with "ECONNREFUSED".
+        if (process.env.NODE_ENV === "test") {
+            return [];
+        }
```
Comment (allisonking) [inline clients/admin-ui/next.config.js:28]: ah, great improvement 👏

**Review by allisonking — APPROVED** [review]
"cypress flakes" sounds like a cereal I'd try

File: .github/workflows/frontend_checks.yml (hunk 2 above)
Comment (ssangervasi) [reply .github/workflows/frontend_checks.yml:76]: They live in the github ether and show up in the action summary:
Screen Shot 2022-11-10 at 10 36 55

That 5.7mb was from a single spec failing, but has all the recordings. I'm sure we can make it smarter if we need to.

**Pull request merged**